content/2017.03.building-projects-as-a-distributed-team.rst

```diff
@@ -0,0 +1,251 @@
+
+You need to explain what you are trying to solve, you need to find out
+how to solve the issue and agree on the solution you are putting
+together, also having feedback will help you to choose the best
+solution even before having written a single line of code.
+
+
+The process of building software in a remote team
+=================================================
+
+Working in a remote team is really close from working with a
+community.
+
+You need to explicit everything you do, you need to find consensuses,
+you need to take feedback, you need to propose solutions and try
+things.
+
+As a team, as a community, you are stronger because instead of being
+alone you can rely on others to help you make the best choices and
+write the best software.
+
+
+The Minimal Viable Prototype (MVP)
+----------------------------------
+
+The first thing to do, is to gather together and brainstorm around the
+project to define what is the **Minimal Viable Prototype**.
+
+At this stage it is really important to start talking about
+prototypes, the project version ``0.1.0`` starts with a ``0`` which
+means that we are still figuring out what we are doing and possibly
+breaking the API, features, expectations.
+
+Everything that is not in the feature set of the MVP is put in a
+Roadmap document for later; to document the fact that we thought about
+it and that we will possibly care about it later.
+
+We write a small article to resume the discussion and to explain what
+is the MVP feature set.
+
+
+Answering complex questions
+---------------------------
+
+Some features of the MVP might not be trivial.
+
+I have a bunch of examples of hard problems:
+
+- How do you handle permissions in Kinto?
+- How do we support multiple database backend?
+- How do we handle ``content-editable`` divs with React or Elm?
+- How do you implement an OAuth flow from a web-extension?
+- How do you handle pluggable notifications in Kinto?
+- How do we configure email notifications in Kinto?
+- How do we enable people to attach attachment to records in Kinto?
+- How do we get Firefox Account keys derivated from the password from
+  an Oauth flow while we never get in touch with the password?
+
+Each problem will have to be addressed by someone and presented to the
+team members so that they can comment on it. Sometimes the way to go
+is obvious, sometimes there is only one good solution, but sometimes
+the good solution is not obvious.
+
+When a complex problem arise, what we want is to learn from it and to
+share what we've learn. A good way of doing so it to write a Blue
+print article exposing pros and cons for the problem and explaining
+which solution was chosen, why and how we've implemented it.
+
+
+Conclusion
+==========
+
+Make the process of building software a way to learn new things and
+share what you've learn. Then you don't have to worry about your
+project to be decommissioned because you've already got the take aways
+of building it.
```

Comment:
We could also mention the fact that Kinto core has been extracted from the Readinglist codebase somewere in the post.